packages/contracts/src/plugins/counter-example/v1/CounterV1PluginSetup.sol

```diff
@@ -18,13 +18,11 @@ contract CounterV1PluginSetup is PluginSetup {
 
     // For testing purposes, the below are public...
     MultiplyHelper public multiplyHelperBase;
-    CounterV1 public counterBase;
 
     address private constant NO_CONDITION = address(0);
 
-    constructor() {
-        multiplyHelperBase = new MultiplyHelper();
-        counterBase = new CounterV1();
+    constructor(MultiplyHelper _helper) PluginSetup(address(new CounterV1())) {
+        multiplyHelperBase = _helper;
     }
```

Comment:
I like the former scenario better, because it is explicit about what the developer needs to do.

Using `PluginSetup(...)` before the constructor, things appear less obvious to a newcomer, IMO.

Reply:
It has some advantages, although not ground-breaking ones:

Devs need to implement one functions less. If they don't put an implementation address in the `PluginSetup ` constructor, the code won't even compile. This way, they cannot forget. 

Having a `address public immutable implementation` limits the amount of trickery that could be added inside `function getImplementationAddress() external view returns (address)`  (see the discussion https://discord.com/channels/672466989217873929/1076067630152175636/1077163656942456963).
However, this is not perfect as developers don't need to use our `PluginSetup` and can make their own base class fulfilling the `IPluginSetup` interface. So, we cannot be sure, but we could analyze the contracts off-chain and display a warning on the frontend if they deviate